### PR TITLE
Add `snow ws version list` command

### PR DESCRIPTION
**Pull request by sfc-gh-fcampbell**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Adds `snow ws version list` command to be invoked on an app package entity. Behaves the same as `snow app version list`.

### CODE REVIEW

**Review by sfc-gh-melnacouzi — COMMENTED** [review]
Can we please revisit this?
I don't think we should use snow ws version-list. We should use snow ws version list instead.

**Comment by sfc-gh-fcampbell** [comment]
Yeah I'm fine with it, that's what the auto-generation is gonna do anyways

**Review by sfc-gh-cgorrie — DISMISSED** [review]

File: src/snowflake/cli/_plugins/workspace/commands.py [hunk 1]
```diff
@@ -224,3 +224,31 @@ def validate(
         entity_id,
         EntityActions.VALIDATE,
     )
+
+
+version = SnowTyperFactory(
+    name="version",
+    help="Manages versions for project entities.",
```
Comment (sfc-gh-gbloom) [inline src/snowflake/cli/_plugins/workspace/commands.py:231]:
```suggestion
    help="Manages versions for application package entities.",
```
or something similar

**Review by sfc-gh-gbloom — COMMENTED** [review]
Let's add integration test coverage as well?

File: src/snowflake/cli/_plugins/workspace/commands.py (hunk 1 above)
Comment (sfc-gh-fcampbell) [reply src/snowflake/cli/_plugins/workspace/commands.py:231]: this command group isn't specific to just application packages though, we could have other entity types that would support version operations

Comment (sfc-gh-gbloom) [reply src/snowflake/cli/_plugins/workspace/commands.py:231]: Right, my bad. I'm not sure what "project entities" exactly means, but it probably doesn't really matter as we'll soon move the help messages to the concrete actions.

**Review by sfc-gh-melnacouzi — APPROVED** [review]

**Pull request merged**